File: code/statistics/python/stat/graficos/histograma.py
## Histograma de Python

# Paso 1. Importación de bibliotecasnecesarias

from pathlib import Path
import pandas as pd
import matplotlib.pylab as plt

# Paso 2. leyendo data
DATA = Path().resolve() / 'data'

STATE_CSV = DATA / 'state.csv'
state = pd.read_csv(STATE_CSV)

# Paso 3. Histograma

ax = state['Murder.Rate'].plot.hist(density=True, xlim=[0, 12], 
                                    bins=range(1,12), figsize=(4, 4))

# Paso 4. Añadir la densidad de Kernel al histograma
state['Murder.Rate'].plot.density(ax=ax)

# Paso 5. Etiquetar el eje x
ax.set_xlabel('Murder Rate (per 100,000)')

# Paso 6. Ajuste del diseño y visualización del gráfico:

plt.tight_layout()
plt.show()

File: ciencia-de-datos/code/statistics/r/stastics/graficos/histograma.R
## Histograma

# leyendo data
PSDS_PATH <- file.path(getwd())
state <- read.csv(file.path(PSDS_PATH, 'data', 'state.csv'))

breaks <- seq(from=min(state[['Population']]), 
              to=max(state[['Population']]), length=11)

options(scipen=5)
#hist(state[['Population']], breaks=breaks)

# densidad

hist(state[['Murder.Rate']], freq=FALSE )
lines(density(state[['Murder.Rate']]), lwd=3, col='blue')
